File: from-3101-to-3200/3200-maximum-height-of-a-triangle/solution.py
class Solution:
    def maxHeightOfTriangle(self, red: int, blue: int) -> int:
        return max(self._helper(red, blue), self._helper(blue, red))

    def _helper(self, red: int, blue: int) -> int:
        height = 0
        rowToComplete = 1

        while True:
            if rowToComplete % 2 == 1:
                if red < rowToComplete:
                    break
                red -= rowToComplete
            else:
                if blue < rowToComplete:
                    break
                blue -= rowToComplete

            height += 1
            rowToComplete += 1
        return height
        

File: from-3101-to-3200/3200-maximum-height-of-a-triangle/solution.go
func maxHeightOfTriangle(red int, blue int) int {
    redFirst := helper(red, blue)
    blueFirst := helper(blue, red)
    if redFirst > blueFirst {
        return redFirst
    }
    return blueFirst
}

func helper(red int, blue int) int {
    curHeight := 0
    rowToComplete := 1
    for true {
        if rowToComplete % 2 == 1 {
            if red < rowToComplete {
                break
            }
            red -= rowToComplete
        } else {
            if blue < rowToComplete {
                break
            }
            blue -= rowToComplete
        }
        curHeight += 1
        rowToComplete += 1
    }
    return curHeight
}